### PR TITLE
DAOS-3730 tests: Starting agent and server with common params (#2403)

### PR DESCRIPTION
**Pull request by saurabhtandan**
Ensuring the daos_server and daos_agent commands are started by
TestWithServers using configuration files with equal common settings.
These changes also enable configuring the transport credentials of both
the daos_agent and daos_server through the test yaml file.  Since the
default configuration file settings are now stored in classes the
src/tests/ftest/config_file_gen.py utility has been added to replace
the src/tests/ftest/data/daos_*_baseline.yaml files by generating both
files with a common access point and the defaults used by the tests.

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>
Conflicts:
	src/tests/ftest/control/daos_admin_privileged.py
	src/tests/ftest/control/super_block_versioning.py
	src/tests/ftest/data/daos_server_baseline.yaml
	src/tests/ftest/io/nvme_fragmentation.py
	src/tests/ftest/launch.py
	src/tests/ftest/server/metadata.py
	src/tests/ftest/util/agent_utils.py
	src/tests/ftest/util/dfuse_utils.py
	src/tests/ftest/util/fio_test_base.py
	src/tests/ftest/util/ior_test_base.py
	src/tests/ftest/util/ior_utils.py
	src/tests/ftest/util/mdtest_test_base.py
	src/tests/ftest/util/mdtest_utils.py
	src/tests/ftest/util/server_utils.py

### CODE REVIEW

File: src/tests/ftest/util/apricot/apricot/test.py [hunk 1]
```diff
@@ -242,91 +248,307 @@ def __init__(self, *args, **kwargs):
...
+                # partition information to populate the list of hosts.
+                setattr(self, partition_name, partition)
+                setattr(self, host_list_name, get_partition_hosts(partition))
+            elif partition is not None and host_list is not None:
+                self.fail(
+                    "Specifying both a {} partition name and a list of hosts "
+                    "is not supported!".format(name))
+
+        # For API calls include running the agent on the local host
+        self.hostlist_clients = include_local_host(self.hostlist_clients)
```
[the remark this answers is not in the document]
Comment (saurabhtandan) [reply src/tests/ftest/util/apricot/apricot/test.py]: #2564 is already cherry-picked above which has changes for soak.py as well. Was there any other PR you meant by any chance?

[the remark this answers is not in the document]
Comment (phender) [reply src/tests/ftest/util/apricot/apricot/test.py]: No, my comment was made against your first commit changes.  The second commit resolved the issue.

**Comment by phender** [comment]
You will also want to include a cherry-pick of the https://github.com/daos-stack/daos/pull/2682 changes

File: src/tests/ftest/util/server_utils.py [hunk 2]
```diff
@@ -121,10 +121,11 @@ def update_pattern(self, mode, host_qty):
         """
         if mode == "format":
             self.pattern = self.FORMAT_PATTERN
-            self.pattern_count = host_qty
+        elif mode == "reformat":
+            self.pattern = self.REFORMAT_PATTERN
```
Comment (phender) [inline src/tests/ftest/util/server_utils.py:125]: Not sure if pylint will complain about this one.  The `self.REFORMAT_PATTERN` definition was added in https://github.com/daos-stack/daos/pull/2645, but it does not make sense to cherry-pick that as well.  If it is a problem I'd either remove these lines (preferred) or add [this line](https://github.com/daos-stack/daos/blob/dcb7924185c9cc41da80392f9dcda076dcfcbf41/src/tests/ftest/util/server_utils.py#L42).

**Pull request closed without merging**